src/ArmTemplates/Extractor/EntityExtractors/ParametersExtractor.cs

```diff
@@ -268,6 +268,7 @@ async Task AddSecretValuesParameters()
             if (extractorParameters.ParameterizeBackend)
             {
                 parameters.Add(ParameterNames.BackendSettings, new TemplateObjectParameterProperties() { Value = backendResources.BackendNameParametersCache });
+                parameters.Add(ParameterNames.BackendProxy, new TemplateObjectParameterProperties() { Value = backendResources.BackendProxyParametersCache });
```

Comment:
just to clarify:
- we are building a `BackendProxyParametersCache` dictionary with all possible combinations in `BackendExtractor`
- here we are including the dictionary to parameters file, and it will be exposed to user for modification

yep?

Reply:
Yes, that is correct. 
BackendProxyParametersCache object is populated in BackendExtractor and then passed to the parameters file which could be modified.